src/service/troopMovements.ts

```diff
@@ -0,0 +1,186 @@
+import { UnitChange } from "../entity/UnitChange";
+import { CountryData } from "../wrapper/models/country";
+import { Client, TextChannel, MessageEmbed } from "discord.js";
+import { AppDataSource } from "..";
+import { Guild } from "../entity/Guild";
+import { getIcon, convertAxisControl, getColor } from "./util/team";
+import { Color } from "./util/constants";
+import { getDistance } from "../map/util";
+
+type Units = Omit<UnitChange, "id">;
+
+export function changedUnitsFromWorld(
+    world: CountryData[],
+    prevUnits: UnitChange[]
+): Units[] {
+    return world
+        .map((country) => {
+            const prev = prevUnits.find(
+                (units) => parseInt(country.id) === units.country
+            );
+            if (prev && !compareUnits(country, prev)) {
+                return buildUnitChange(country, prev);
+            }
+            return;
+        })
+        .filter((a) => a) as Units[];
+}
+
+export function detectOrigin(changedWorld: Units[]): Units[] {
+    const allies: Map<number, Units[]> = new Map();
+    const axis: Map<number, Units[]> = new Map();
+
+    changedWorld.map((c1) => {
+        if (c1.delta_allies !== 0) {
+            const delta = Math.abs(c1.delta_allies);
+            if (allies.has(delta)) {
+                allies.get(delta)?.push(c1);
+            } else {
+                allies.set(delta, [c1]);
+            }
+        }
+
+        if (c1.delta_axis !== 0) {
+            const delta = Math.abs(c1.delta_axis);
+            if (axis.has(delta)) {
+                axis.get(delta)?.push(c1);
+            } else {
+                axis.set(delta, [c1]);
+            }
+        }
+    });
+
+    return buildAlliesChanges(allies).concat(buildAxisChanges(axis));
+}
+
+export async function prepareAndSendMessage(
+    client: Client,
+    changedUnits: Units[],
+    world: CountryData[]
+) {
+    const countryInfo: Map<number, CountryData> = new Map();
+    world.map((country) => {
```

Comment:
this should be forEach, but it would be much better to use the `Map` constructor instead.
```ts
const contryInfo = new Map(
    world.map((country) => [parseInt(country.id), country])
);
```